node-test/src/test/scala/org/bitcoins/node/NeutrinoNodeWithWalletTest.scala

```diff
@@ -0,0 +1,111 @@
+import org.bitcoins.rpc.client.common.BitcoindVersion
+import org.bitcoins.server.BitcoinSAppConfig
+import org.bitcoins.testkit.BitcoinSTestAppConfig
+import org.bitcoins.testkit.fixtures.UsesExperimentalBitcoind
+import org.bitcoins.testkit.node.NodeUnitTest.NeutrinoNodeFundedWalletBitcoind
+import org.bitcoins.testkit.node.{NodeTestUtil, NodeUnitTest}
+import org.bitcoins.wallet.api.UnlockedWalletApi
+import org.scalatest.exceptions.TestFailedException
+import org.scalatest.{DoNotDiscover, FutureOutcome}
+
+import scala.concurrent.duration._
+import scala.concurrent.{Future, Promise}
+
+@DoNotDiscover
+class NeutrinoNodeWithWalletTest extends NodeUnitTest {
+
+  /** Wallet config with data directory set to user temp directory */
+  implicit override protected def config: BitcoinSAppConfig =
+    BitcoinSTestAppConfig.getNeutrinoTestConfig()
+
+  override type FixtureParam = NeutrinoNodeFundedWalletBitcoind
+
+  def withFixture(test: OneArgAsyncTest): FutureOutcome = {
+    withNeutrinoNodeFundedWalletBitcoind(test,
+                                         callbacks,
+                                         Some(BitcoindVersion.Experimental))
+  }
+
+  private val assertionP: Promise[Boolean] = Promise()
+
+  private val walletP: Promise[UnlockedWalletApi] = Promise()
+  private val walletF: Future[UnlockedWalletApi] = walletP.future
+
+  val amountFromBitcoind = 1.bitcoin
+
+  def callbacks: SpvNodeCallbacks = {
+    val onBlock: DataMessageHandler.OnBlockReceived = { block =>
+      for {
+        wallet <- walletF
+      } yield {
+        for {
+          prevBalance <- wallet.getUnconfirmedBalance()
+          _ <- wallet.processBlock(block, confirmations = 0)
+          balance <- wallet.getUnconfirmedBalance()
+        } yield {
+          val result = balance == prevBalance + amountFromBitcoind
+          assertionP.success(result)
+        }
+      }
+    }
+    SpvNodeCallbacks(
+      onBlockReceived = Seq(onBlock)
+    )
+  }
+
+  it must "rescan and receive information about received payments" taggedAs (UsesExperimentalBitcoind) in {
+    param =>
+      val NeutrinoNodeFundedWalletBitcoind(node, wallet, bitcoind) = param
+
+      walletP.success(wallet)
+
+      var cancellable: Option[Cancellable] = None
+
+      def setTimeout(tx: DoubleSha256DigestBE): DoubleSha256DigestBE = {
+        // how long we're waiting for a tx notify before failing the test
+        val delay = 25.seconds
+
+        val failTest: Runnable = new Runnable {
+          override def run = {
+            if (!assertionP.isCompleted) {
+              val msg =
+                s"Did not receive a block within $delay"
+              logger.error(msg)
+              assertionP.failure(new TestFailedException(msg, 0))
+            }
+          }
+        }
+
+        logger.debug(s"Setting timeout for receiving TX through node")
+        cancellable = Some(system.scheduler.scheduleOnce(delay, failTest))
+        tx
+      }
+
+      for {
```

Comment:
So it seems to me the test case should 

1. Send money to the bitcoin-s wallet
2. Confirm that money on the bitcoin network
3. Erase the `SpendingInfoTable` 
4. Read the addresses we have generated from the `AddressTable` 
5. Rescan the chain against those SPKs we read from the `AddressTable`
6. For the scripts that matched, insert them into `SpendingInfoTable`.
7. Assert we have the same amount of money as we had in step 2. 
8. Assert we have the same contents in `SpendingInfoTable` as we had in step 3.


This seems to be what we want to do, and it does simplify the problem of generating addresses from an HDKey but avoiding it all together but just assuming `AddressTable` is populated. In the future, we will have to populate `AddressTable` from the HDKey, but we can do that on a separate PR.

Reply:
I think the name on this test case shouldn't include anything about rescanning as it is just about receiving money. I would rename the current `rescan` method to `fetchBlock()` or something like that.

This starts to tie into what we talked about on Friday with moving filters into the wallet/chain project and also having rescan logic live there. This should simplify operations in the node project. Let me know if you want me to clarify this comment.